core/state_processor_rip7560.go

```diff
@@ -230,9 +229,12 @@ func refundPayer(vpr *ValidationPhaseResult, state vm.StateDB, gasUsed uint64, g
 // CheckNonceRip7560 prechecks nonce of transaction.
 // (standard preCheck function check both nonce and no-code of account)
 func CheckNonceRip7560(tx *types.Rip7560AccountAbstractionTx, st *state.StateDB) error {
-	// Make sure this transaction's nonce is correct.
+	// RIP-7712 two-dimensional nonce is checked on-chain
+	if tx.IsRip7712Nonce() {
+		return nil
```

Comment:
In line with checknonce's functionality, wouldn't it make sense to also check for correct request values for two-dimensional nonces?

Reply:
But we cannot check it before we call `NonceManager` contract with the given `NonceKey` and `Nonce` value, and it will induce extra gas in the transaction. Maybe if we want to merge the 2 things in a function, we may make something like `checkAndPrepareNonceValidationMessage` function, and put the functionalities of `CheckNonce` into it. How do you think about this?

Reply:
If the nonceManager contract is at the predeploy level and the contents can be trusted, I think it is possible to calculate the corresponding storage slot, map, and call state.GetState. I think this part can be done later as a separate PR.

Reply:
But I think it can confuse how nonce validation gas is calculated since executing extra `state.Getstate` also consumes computing power of geth so that it should also be counted as an extra gas. Also it is a duplicate, since storage access to that slot must also be done when executing nonce validation message in NonceManager contract.

Reply:
I agree with the above comments to some extent. Hmmm... I was thinking that if we look at 2D-Nonce as a native feature, wouldn't it be nice to provide some sort of getNonce(address, nonceKey) access that is equivalent to the existing getNonce(address). What do you think?

Reply:
If we need it inside the protocol (geth, bundler client), I agree with making that function. But I cannot come up with any cases that need 2D nonce inside the client (maybe helper function inside bundler? idk). Do you have any use cases for the function?